web/services/nft_service.py

```
@@ -0,0 +1,234 @@
+from web3 import Web3
+
+from web.models import Achievement, NFTBadge
+
+logger = logging.getLogger(__name__)
+
+# Blockfrost API Configuration
+
+# Use WEB3_URL_PROVIDER for Polygon RPC URL
+WEB3_URL_PROVIDER = settings.WEB3_PROVIDER_URL
+if not WEB3_URL_PROVIDER:
+    raise ValueError("❌ WEB3_PROVIDER_URL is missing in environment variables")
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)